core/assays.chainmail

```diff
@@ -0,0 +1,91 @@
+/**
+ * Creator and validator of asset Amounts.
+ * 
+ * Amounts are the canonical description of tradable goods. They are manipulated
+ * by mints, and represent the goods and currency carried by purses and
+ * payments. They can be used to represent currency, stock, or the abstract
+ * right to participate in a particular exchange.
+ * 
+ * Assays distinguish between Quantities and Amounts. Quantities describe the
```

Comment:
I just realized that we're missing all such pass-by-copy type declarations. Rather than describing an Amount only with prose, we should have something like

```
struct Amount(Label, Quantity) {
  label :Label;
  quantity: Quantity;
}

struct Label(Description) {
  issuer :Issuer;
  description :Description;
}
```

Reply:
Do Cap'n Proto parameterized types have a way to indicate type constraints? In something like Java Generics, I would have the type parameters above extend `Comparable`.